### PR TITLE
Решение задачи

### PR DESCRIPTION
**Pull request by ser9eevich**
Попытка № 2

### CODE REVIEW

**Comment by msmirnov-c** [comment]
1:4   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   2:56  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   3:82  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   4:3   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   5:36  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   6:22  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   7:9   error    'resultArr' is never reassigned. Use 'const' instead                          prefer-const
   7:24  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   8:22  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
   9:22  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  10:1   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  11:22  error    Expected '===' and instead saw '=='                                           eqeqeq
  11:27  error    Missing space before opening brace                                            space-before-blocks
  11:28  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  12:32  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  13:9   error    Expected blank line before this statement                                     padding-line-between-statements
  13:26  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  14:6   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  15:1   error    Trailing spaces not allowed                                                   no-trailing-spaces
  15:5   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  16:19  error    Unexpected function expression                                                prefer-arrow-callback
  16:19  warning  Unexpected unnamed function                                                   func-names
  16:27  error    Missing space before function parentheses                                     space-before-function-paren
  16:34  error    'i' is defined but never used                                                 no-unused-vars
  16:37  error    'arr' is defined but never used                                               no-unused-vars
  16:43  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  17:1   error    Expected indentation of 8 spaces but found 7                                  indent
  17:8   error    Expected { after 'if' condition                                               curly
  17:12  error    Unary word operator 'typeof' must be followed by whitespace                   space-unary-ops
  17:29  error    Strings must use singlequote                                                  quotes
  17:38  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  18:1   error    Expected indentation of 12 spaces but found 7                                 indent
  18:8   error    Expected no linebreak before this statement                                   nonblock-statement-body-position
  18:24  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  19:1   error    Expected indentation of 8 spaces but found 7                                  indent
  19:8   error    Expected { after 'else'                                                       curly
  19:12  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  20:1   error    Expected indentation of 12 spaces but found 7                                 indent
  20:8   error    Expected no linebreak before this statement                                   nonblock-statement-body-position
  20:20  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  21:8   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  22:1   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  23:23  error    Missing space before opening brace                                            space-before-blocks
  23:24  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  24:21  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  25:5   error    Closing curly brace does not appear on the same line as the subsequent block  brace-style
  25:6   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  26:5   error    Expected space(s) after "else"                                                keyword-spacing
  26:9   error    Unnecessary 'else' after 'return'                                             no-else-return
  26:10  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  27:24  error    Missing radix parameter                                                       radix
  27:42  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  28:32  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  29:9   error    'resultStr' is not defined                                                    no-undef
  29:42  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  30:1   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  31:29  error    'resultStr' is not defined                                                    no-undef
  31:53  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  32:1   error    Expected indentation of 12 spaces but found 8                                 indent
  32:24  error    Missing radix parameter                                                       radix
  32:33  error    'resultStr' is not defined                                                    no-undef
  32:47  error    Missing semicolon                                                             semi
  32:47  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  33:1   error    Expected indentation of 8 spaces but found 4                                  indent
  33:6   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  34:6   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  35:5   error    Expected blank line before this statement                                     padding-line-between-statements
  35:22  error    Trailing spaces not allowed                                                   no-trailing-spaces
  35:25  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  36:2   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  37:1   error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  38:19  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  39:14  error    Missing trailing comma                                                        comma-dangle
  39:14  error    Expected linebreaks to be 'LF' but found 'CRLF'                               linebreak-style
  40:3   error    Newline required at end of file but not found                                 eol-last

✖ 75 problems (74 errors, 1 warning)
  66 errors and 0 warnings potentially fixable with the `--fix` option.

File: index.js [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+    let fail = false;
+
+    if (array.length === 0) {
+        resultArr.push(number);
+
+        return resultArr;
+    }
+
+    array.forEach((item) => {
+        if (typeof (item) === 'number') origStr += item;
```
Comment (ghost) [inline index.js]: Лучше провалидировать массив заранее, а тут можно использовать другой метод массива. Даже два.

File: index.js [hunk 2]
```diff
@@ -1,11 +1,40 @@
...
+        else fail = true;
+    });
+
+    if (fail === true) {
```
Comment (ghost) [inline index.js]: Нет смысла булеву переменную проверять в явном виде, если вы знаете, что там точно булево значение.

File: index.js [hunk 3]
```diff
@@ -1,11 +1,40 @@
...
+/**
+ * @param array – массив элементов. Например, [1, 8, 3]
+ * @param number – число, которое нужно прибавить к числу, полученному из массива
+*/
+function addNumber(array, number) {
+    let origStr = '';
+    const resultArr = [];
+    let resultNumber;
+    let resultStr = '';
+    let fail = false;
```
Comment (ghost) [inline index.js]: Эта переменная лишняя, нужно переписать без нее.

File: index.js [hunk 4]
```diff
@@ -1,11 +1,40 @@
...
+    array.forEach((item) => {
+        if (typeof (item) === 'number') origStr += item;
+        else fail = true;
+    });
+
+    if (fail === true) {
+        return null;
+    }
+
+    resultNumber = parseInt(origStr, 10);
```
Comment (ghost) [inline index.js]: Можно объявлять переменную прямо здесь, а не выше.

File: index.js [hunk 5]
```diff
@@ -1,11 +1,40 @@
...
+    resultNumber += number;
+    resultStr = String(resultNumber);
```
Comment (ghost) [inline index.js]: Аналогично можно объявить здесь.

File: index.js [hunk 6]
```diff
@@ -1,11 +1,40 @@
...
+
+    for (let i = 0; i < resultStr.length; i++) {
```
Comment (ghost) [inline index.js]: Посмотрите на метод map.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by ser9eevich** [comment]
Исправил весь код, вроде работает

File: index.js [hunk 7]
```diff
@@ -1,11 +1,29 @@
...
-    addNumber
-};
\ No newline at end of file
+/**
+ * @param array – массив элементов. Например, [1, 8, 3]
+ * @param number – число, которое нужно прибавить к числу, полученному из массива
+*/
+function addNumber(array, number) {
+    const checkArr = array.every((item) => typeof (item) === 'number');
+    if (checkArr) {
```
Comment (ghost) [inline index.js]: Если взять обратное условие, то можно не делать такую логику внутри условия, а вернуть тут null.
Так лучше читаемость.

File: index.js [hunk 8]
```diff
@@ -1,11 +1,29 @@
...
-module.exports = {
-    addNumber
-};
\ No newline at end of file
+/**
+ * @param array – массив элементов. Например, [1, 8, 3]
+ * @param number – число, которое нужно прибавить к числу, полученному из массива
+*/
+function addNumber(array, number) {
+    const checkArr = array.every((item) => typeof (item) === 'number');
```
Comment (ghost) [inline index.js]: Название не помогает понять, что проверяли. Например, то, что каждый элемент число.

File: index.js [hunk 9]
```diff
@@ -1,11 +1,29 @@
...
+    if (checkArr) {
+        let resultArr = [];
+        if (array.length === 0) {
```
Comment (ghost) [inline index.js]: Такой ситуации не предусмотрено. Наверное, даже должен быть null даже в таком случае, так как мы прибавляем число к "пустоте".

File: index.js [hunk 10]
```diff
@@ -1,11 +1,29 @@
...
+            resultArr.push(number);
+
+            return resultArr;
```
Comment (ghost) [inline index.js]: Можно просто вернуть `[number]`;

File: index.js [hunk 11]
```diff
@@ -1,11 +1,29 @@
...
-};
\ No newline at end of file
+/**
+ * @param array – массив элементов. Например, [1, 8, 3]
+ * @param number – число, которое нужно прибавить к числу, полученному из массива
+*/
+function addNumber(array, number) {
+    const checkArr = array.every((item) => typeof (item) === 'number');
+    if (checkArr) {
+        let resultArr = [];
```
Comment (ghost) [inline index.js]: Объявление переменной можно перенести вниз.

File: index.js [hunk 12]
```diff
@@ -1,11 +1,29 @@
...
+        if (array.length === 0) {
+            resultArr.push(number);
+
+            return resultArr;
+        }
+
+        let origNum = parseInt(array.join(''), 10);
+        origNum += number;
```
Comment (ghost) [inline index.js]: Можно и сразу прибавить. Тогда переменная будет const.

File: index.js [hunk 13]
```diff
@@ -1,11 +1,29 @@
...
+        const numStr = origNum.toString();
+        resultArr = numStr.split('');
+        resultArr = resultArr.map((item) => parseInt((item), 10));
```
Comment (ghost) [inline index.js]: Можно на предыдущую строку. Либо на предыдущей строке создать новую переменную с осмысленным названием. Но вот так перезаписывать не нужно. Получается, что в разные моменты времени в переменной содержатся разные по смыслу значения.

**Review by ghost — CHANGES_REQUESTED** [review]

File: index.js (hunk 10 above)
Comment (ser9eevich) [reply index.js]: @AlexNg9 Но по условию метод должен возвращать массив, а если мы вернем `[number]`, то получается решение задачи будет не верным?

File: index.js [hunk 14]
```diff
@@ -1,11 +1,21 @@
...
-    addNumber
-};
\ No newline at end of file
+/**
+ * @param array – массив элементов. Например, [1, 8, 3]
+ * @param number – число, которое нужно прибавить к числу, полученному из массива
+*/
+function addNumber(array, number) {
+    const allElementsAreNumbers = array.every((item) => typeof (item) === 'number');
+    if (allElementsAreNumbers && array.length !== 0) {
```
Comment (ghost) [inline index.js]: Лучше взять отрицание этого условия и вернуть в if null.

File: index.js [hunk 15]
```diff
@@ -1,11 +1,21 @@
...
+        const origNum = parseInt(array.join(''), 10) + number;
+        const numStr = origNum.toString();
+        const stringArr = numStr.split('');
```
Comment (ghost) [inline index.js]: Лучше подробнее называть. `digitStringArray`, например.

File: index.js [hunk 16]
```diff
@@ -1,11 +1,21 @@
...
+        const resultArr = stringArr.map((item) => parseInt((item), 10));
```
Comment (ghost) [inline index.js]: Можно сразу вернуть эту строку, не заводя переменную.

**Review by ghost — CHANGES_REQUESTED** [review]

File: index.js (hunk 16 above)
Comment (ser9eevich) [reply index.js]: Я хотел так сделать, но думал что это неправильно

File: index.js (hunk 14 above)
Comment (ser9eevich) [reply index.js]: Можете объяснить почему так лучше? Это если массив попадется не удовлетворяющий условию, сразу вернуть `null`, а не проходить весь код?

File: index.js (hunk 16 above)
Comment (ser9eevich) [reply index.js]: После данного исправления, вылезли ошибки  

> 13:5   error  Return statement should not contain assignment       no-return-assign
> 13:12  error  'resultArr' is not defined                           no-undef
> 13:38  error  Expected parentheses around arrow function argument  arrow-parens

**Review by ghost — APPROVED** [review]
👍

**Pull request closed without merging**